File: javaaa/Encapsulation.java
package javaaa;

class EncapsulatedProduct {
    private String name;
    private double price;

    EncapsulatedProduct(String name, double price) {
        this.name = name;
        this.price = price;
    }

    String getName() {
        return name; 
    }

    double getPrice() { 
        return price; 
    }
}

public class Encapsulation {
    public static void main(String[] args) {
        EncapsulatedProduct p = new EncapsulatedProduct("Laptop", 2000);
        System.out.println(p.getName() + ": $" + p.getPrice());
    }
}

File: darttt/Encapsulation.dart
class EncapsulatedProduct {
  String _name;   
  double _price; 

  EncapsulatedProduct(this._name, this._price);

  String get name => _name;
  double get price => _price;
}

void main() {
  var p = EncapsulatedProduct("Laptop", 2000);
  print("${p.name}: \$${p.price}");
}
